src/components/choose-locale/change-locale.component.tsx

```diff
@@ -0,0 +1,49 @@
+import { refetchCurrentUser } from "@openmrs/esm-api";
+import React, { useEffect, useState } from "react";
+import { updateUserProperties } from "./change-locale.resource";
+
+interface ChangeLocaleProps {
+  allowedLocales: Array<string>;
+  user: any;
+}
+
+export const ChangeLocale: React.FC<ChangeLocaleProps> = ({
+  allowedLocales,
+  user
+}) => {
+  const [userProps, setUserProps] = useState(user.userProperties);
+  const options = allowedLocales?.map(locale => (
+    <option value={locale} key={locale}>
+      {locale}
+    </option>
+  ));
+
+  useEffect(() => {
+    if (user.userProperties.defaultLocale !== userProps.defaultLocale) {
+      updateUserProperties(user.uuid, userProps, new AbortController()).then(
+        response => {
+          if (response.ok) {
+            refetchCurrentUser();
+          }
+        }
+      );
+    }
+  }, [userProps]);
+
+  return (
+    <div className="omrs-margin-12">
+      <label htmlFor="selectLocale">
+        <span>Select Locale</span>
```

Comment:
Let's not forget that we still need to internationalize primary-navigation.

Reply:
@brandones I thought for this specific label we should default to `en`. If I'm a user and If by mistake change locale to `fr` that I don't understand, I can understand my way back to revert this, if this label is not translated. 

See below(Reference Application):
![screencapture-openmrs-spa-org-openmrs-adminui-myaccount-myAccount-page-2020-10-05-22_12_46](https://user-images.githubusercontent.com/26084581/95121783-36a17700-0758-11eb-887f-7a700deef0b5.png)

I thought the above is intentionally untranslated?

Reply:
Oh, interesting thought. Sure, let's leave it like that then.